projects/stargazer/plugins/other/radius/radius.cpp

```diff
@@ -17,3 +23,48 @@ RADIUS::RADIUS()
 {
 }
 
+int RADIUS::Start()
+{
+    m_thread = std::jthread([this](auto token){ Run(std::move(token)); });
+    return 0;
+}
+
+int RADIUS::Stop()
+{
+    std::lock_guard lock(m_mutex);
+
+    if (!m_thread.joinable())
+        return 0;
+
+    m_thread.request_stop();
+
+
+    if (isRunning)
+        m_thread.detach();
+    else
+        m_thread.join();
+
+    return 0;
+}
+
+int RADIUS::Run(std::stop_token token)
+{
+    std::lock_guard lock(m_mutex);
+    isRunning = true;
+
+    while (!token.stop_requested())
+    {
+        try
+        {
+            boost::asio::io_service io_service;
+            Server server(io_service, "secret", 1812, "/usr/share/freeradius/dictionary");
+            io_service.run();
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Exception: " << e.what() <<"\n";
```

Comment:
Avoid using `std::cout` and `std::cerr`. See how the other plugins report errors.